src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -114,16 +128,31 @@ private IEnumerable<CompletionItem> GetDeclarationTypeCompletions(BicepCompletio
             return Enumerable.Empty<CompletionItem>();
         }
 
+        private IEnumerable<CompletionItem> GetResourceTypeCompletions(SemanticModel model, BicepCompletionContext context)
+        {
+            if (context.Kind != BicepCompletionContextKind.ResourceType)
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            // we need to ensure that Microsoft.Compute/virtualMachines@whatever comes before Microsoft.Compute/virtualMachines/extensions@whatever
+            // similarly, newest api versions should be shown first
+            return model.ResourceTypeProvider.GetAvailableTypes()
+                .OrderBy(rt => rt.FullyQualifiedType, StringComparer.OrdinalIgnoreCase)
```

Comment:
Worth creating a custom `IComparer<>` for this? This is a pretty extreme nitpick, but ordering apiVersion descending will end up with `2020-01-01-preview` (longer string with same prefix) being listed before `2020-01-01`. We have [this implementation](https://msazure.visualstudio.com/One/_git/AzureUX-ARM?path=%2Fsrc%2Fcommon%2FCore%2FComparers%2FApiVersionComparer.cs&_a=contents&version=GBmaster) in ARM.

Reply:
Yeah, I was thinking about it when I was writing that code. At the time I decided to ignore it because it's rare to have an API version with the same date and different suffixes. I have a more readable comparer in the metadata repo (it assigns clear weights to each suffix value). I can pull that in.